dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,733 @@
+                        )
+
+                        vis[freq_sel, ii, :] += np.sum(
+                            pscale * pcnt * pflag * pvis * fringestop_phase, axis=1
+                        )
+                        var[freq_sel, ii, :] += np.sum(
+                            (pscale * pcnt) ** 2 * pflag * pvar, axis=1
+                        )
+                        counter[freq_sel, ii, :] += np.sum(
+                            pscale * pcnt * pflag, axis=1
+                        )
+
+                    self.logger.info(
+                        "Took %0.1f seconds to process this block."
+                        % (time.time() - t0,)
+                    )
+
+                    del bdata
+                    gc.collect()
+
+                # Normalize
+                inv_counter = tools.invert_no_zero(counter)
+                vis *= inv_counter
+                var *= inv_counter ** 2
+
+                ra = np.degrees(np.unwrap(ra))
+                ha = ra - np.degrees(src_ra)
+
+                # Fit response to model
+                if self.perform_fit:
+
+                    fwhm = np.zeros((nfreq, npol), dtype=np.float32)
+                    for ii in range(npol):
+                        fwhm[:, ii] = cal_utils.guess_fwhm(
+                            data.freq, pol=pols[ii][0], dec=src_dec, sigma=True
+                        )
+
+                    flag = counter > 0.0
+
+                    parameter, parameter_cov, resid_rms = fit_point_source_transit(
+                        ha, vis.real, np.sqrt(var), flag=flag, fwhm=fwhm
+                    )
+                # Write to file
+                output_file = os.path.join(
+                    self.write_dir,
+                    "%s_csd_%d_%s.h5" % (src.lower(), csd, self.output_suffix),
+                )
+                self.logger.info("Writing output files...")
+
+                with h5py.File(output_file, "w") as handler:
+
+                    index_map = handler.create_group("index_map")
+                    index_map.create_dataset("freq", data=data.index_map["freq"][:])
+                    index_map.create_dataset("pol", data=np.string_(pols))
+                    index_map.create_dataset("time", data=data.time)
+                    index_map.create_dataset("ra", data=ra)
+                    index_map.create_dataset("ha", data=ha)
+
+                    dset = handler.create_dataset("vis", data=vis)
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    dset = handler.create_dataset(
+                        "weight", data=tools.invert_no_zero(var)
+                    )
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    dset = handler.create_dataset("count", data=counter.astype(np.int))
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    if self.perform_fit:
+
```

Comment:
If you use `ch_util.cal_utils` for performing the fit, then

- `parameter` dataset is saved in the `fitter.param` attribute.
- `parameter_cov` dataset is saved in the `fitter.param_cov` attribute.
- `param` index map is saved in the `fitter.parameter_names` attribute.
- You might consider adding the `chisq` and `ndof` attributes as additional datasets.
- We should probably save our best estimate of the spectrum as a separate dataset, so that theremin does not have to derive it from the beamformed visbilities or model fit.  If you want to use the peak value, then this would be given by `fitter.peak()`.  If you want to use the value at transit, this would be given by `fitter.predict(0.0)`.  These are different due to the centroid wander, and which one is used might be a config parameter.

I would also save some info about the model that was used, so that the parameters saved to the file can be used to make model predictions in the future:

```
        handler.attrs["model_kwargs"] = json.dumps(fitter.model_kwargs)
        handler.attrs["model_class"] = ".".join(
            [getattr(cal_utils.FitGaussAmpPolyPhase, key) for key in ["__module__", "__name__"]]
        )
```